backend/handlers/institution_parent_request_handler.py

```diff
@@ -33,11 +33,13 @@ def put(self, user, request_key):
         request.put()
 
         parent_institution = request.institution_requested_key.get()
-        parent_institution.children_institutions.append(request.institution_key)
-        parent_institution.put()
+
+        if request.institution_key not in parent_institution.children_institutions:
```

Comment:
institution.py's add_child method verifies this condition and add the child. Use it